lib/runner/extract-runnable-items.js

```diff
@@ -0,0 +1,172 @@
+var _ = require('lodash'),
+    sdk = require('postman-collection'),
+    ItemGroup = sdk.ItemGroup,
+    Item = sdk.Item,
+
+    /**
+     * Looks for an Item or ItemGroup matching the id on a collection.
+     *
+     * @param {Collection} collection
+     * @param {String} id
+     *
+     * @returns {Item|ItemGroup|undefined}
+     *
+     * @todo: Possibly add oneDeepById to sdj.ItemGroup?
+     */
+    findItemOrGroupById = function (collection, id) {
+        var entry,
+
+            /**
+             * Iteratee for looking up an element with given id on an item group.
+             *
+             * @param {ItemGroup} itemGroup
+             * @param {String} id
+             *
+             * @returns {Boolean|undefined}
+             */
+            idLookup = function (itemGroup, id) {
```

Comment:
anti-functional

Reply:
Couldn't get to a functional way to do this without shenanigans. Staying with synchronous flag based find now.